shadows/framework/src/main/java/org/robolectric/shadows/ShadowAccessibilityManager.java

```diff
@@ -95,17 +99,23 @@ private static AccessibilityManager createInstance(Context context) throws Excep
   @Implementation
   protected boolean addAccessibilityStateChangeListener(
       AccessibilityManager.AccessibilityStateChangeListener listener) {
+    addAccessibilityStateChangeListener(listener, null);
     return true;
   }
 
   @Implementation(minSdk = O)
   protected void addAccessibilityStateChangeListener(
-      AccessibilityManager.AccessibilityStateChangeListener listener, Handler handler) {}
+      AccessibilityManager.AccessibilityStateChangeListener listener, Handler handler) {
+    onAccessibilityStateChangeListeners.put(listener, handler);
+  }
 
   @Implementation
   protected boolean removeAccessibilityStateChangeListener(
       AccessibilityManager.AccessibilityStateChangeListener listener) {
-    return true;
+    if (listener == null) {
+      throw new IllegalArgumentException("Invalid AccessibilityStateChangeListener");
```

Comment:
Looks like https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/view/accessibility/AccessibilityManager.java;l=852-859?q=removeAccessibilityStateChangeListener&ss=android, https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/util/ArrayMap.java;l=725-732;drc=master, https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/util/ArrayMap.java;l=433-436;drc=master will not throw `IllegalArgumentException` if listener is null.

Reply:
Good catch, fixed!